pgmpy/factors/discrete/DiscreteFactor.py

```diff
@@ -1026,7 +1026,7 @@ def __hash__(self):
             phi.values = phi.values.swapaxes(axis, exchange_index)
         return hash(
             str(sorted_var_hashes)
-            + str(phi.values)
+            + str(hash(phi.values.data.tobytes()))
             + str(phi.cardinality)
```

Comment:
Can also convert `phi.cardinality` to bytes.

Reply:
I am a new Github user, although signed up early. I modify the code according to your suggestion, and commit again, is this right?

Reply:
Yes, this is good.